src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -391,23 +390,43 @@ internal static unsafe void FixupMethodCell(IntPtr hModule, MethodFixupCell* pCe
         }
 
 #if TARGET_WINDOWS
-        private static unsafe IntPtr GetProcAddressWithSuffix(IntPtr hModule, byte* methodName, byte suffix)
+#if TARGET_X86
+        private static unsafe IntPtr GetProcAddressWithMangling(IntPtr hModule, byte* methodName, MethodFixupCell* pCell)
         {
-            int nameLength = string.strlen(methodName);
-
-            // We need to add an extra byte for the suffix, and an extra byte for the null terminator
-            byte* probedMethodName = stackalloc byte[nameLength + 2];
-
-            for (int i = 0; i < nameLength; i++)
+            IntPtr pMethod = Interop.Kernel32.GetProcAddress(hModule, methodName);
+            if (pMethod == IntPtr.Zero && pCell->SignatureBytes >= 0)
```

Comment:
```suggestion
            IntPtr pMethod = Interop.Kernel32.GetProcAddress(hModule, methodName);

#if TARGET_X86
            if (pMethod == IntPtr.Zero && pCell->SignatureBytes >= 0)
```
Nit: The equivalent unmanaged code in CoreCLR has the ifdef in the implementation like this. Fewer lines.